### PR TITLE
[Java] Add support for JSR310 in Jersey Clients

### PR DESCRIPTION
**Pull request by ant3**
This is further (belated) work on #3065 

The Jackson module for JSR310 was incorrectly named and generated code did not compile.  This PR fixes that as well as improving the formatting in the mustache templates (as suggested in previous PR).

### CODE REVIEW

**Comment by wing328** [comment]
I tested both jersey1 and jersey2 Java API client and both compiled without issue

**Comment by wing328** [comment]
@ant3 for your upcoming PRs, please run the following to update the Petstore sample for Java (jersey1, jersey2) Petstore clients:
```
./bin/java-petstore.sh
./bin/java-petstore-jersey2.sh
```
so that the changes will be tested by travis-ci

**Comment by ant3** [comment]
Thanks for merging this so quickly.  The 2 scripts you mentioned don't specify dateLibrary=java8 so wouldn't have picked up the issue.  I could add a java8-petstore-jersey2.sh to test this if you want.

**Comment by wing328** [comment]
Yes, we probably need another script to test dateLibrary option. 

It's still a good practice to generate the Java Petstore for default dateLibrary option so as to ensure the default option is not impacted by the change.

File: modules/swagger-codegen/src/main/resources/Java/ApiClient.mustache [hunk 1]
```diff
@@ -66,7 +66,7 @@ public class ApiClient {
     objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
     objectMapper.enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);
     objectMapper.enable(DeserializationFeature.READ_ENUMS_USING_TO_STRING);{{#java8}}
-    objectMapper.registerModule(new JavaTimeModule());{{/java8}}{{^java8}}
+    objectMapper.registerModule(new JSR310Module());{{/java8}}{{^java8}}
```
Comment (cbornet) [inline modules/swagger-codegen/src/main/resources/Java/ApiClient.mustache]: Why did you use JSR310Module ? It's deprecated and the one to use is JavaTimeModule !

Comment (ant3) [inline modules/swagger-codegen/src/main/resources/Java/ApiClient.mustache]: The generated pom used a slightly older version of the jackson JSR310 library - which didn't have the newer JavaTimeModule.  However, I take your point and have updated the dependency and the code in my latest PR.

**Comment by ant3** [comment]
I have created PR #3186 to address issues raised in comments above.

Thanks for bearing with me.

**Pull request merged**